x-pack/plugins/canvas/public/state/selectors/workpad.js

```diff
@@ -89,8 +89,9 @@ export function getElementCounts(state) {
 
   Object.keys(resolvedArgs).forEach(resolvedArg => {
     const arg = resolvedArgs[resolvedArg];
-    const { expressionRenderable } = arg;
-    const { value, state } = expressionRenderable;
+    const { expressionRenderable, expressionContext } = arg;
+    const expressionObject = expressionRenderable || expressionContext;
```

Comment:
`expressionRenderable` and `expressionContext` are not interchangeable.

I'm a little rusty on this, but I believe `expressionContext` is actually the context that was passed into the expression at that point, so what you're actually getting here is the state of the expression before the function in question. We use that to cache context for the sidebar, so doing things that don't require a data change, like setting the color palette as one example, can happen quickly (we can use the cached context to avoid a round-trip). 

An element should always end up with a `expressionRenderable` value. The time filter eventually gets one, even though it happens to get `expressionContext` first. There's also a subtle bug there, the order is not actually dependable, it's all variable timing.